modules/core/src/main/scala/com.snowplowanalytics.snowplow.snowflake/processing/SnowflakeRetrying.scala

```diff
@@ -54,6 +59,13 @@ object SnowflakeRetrying {
       sql.show
   }
 
+  implicit class FilteredSnowflakeSQLException(val sse: SnowflakeSQLException) extends AnyVal {
+    def isEligibleSetupError: Boolean = {
+      lazy val pattern = "^SQL compilation error:\nTable '(\\w+)' already exists, but current role has no privileges on it.*".r
+      sse.getErrorCode === 3041 && sse.getSQLState === "42710" && pattern.matches(sse.getMessage)
```

Comment:
```suggestion
      pattern.matches(sse.getMessage)
```
I would not be so precise. I think that matching on the error message is enough, for 2 reasons:
- What if Snowflake keeps the same error message, but changes the error code? Very unlikely I know
- Are we 100% sure that no other error code can lead to this error message?

Reply:
Oh, my instinct was to suggest the exact opposite!!  I would match on the error code only, because I think the message is more likely to change over time, whereas error codes tend to stay the same over time.

I noticed [this code](https://github.com/snowflakedb/SnowAlert/blob/b98d7566d120a1cbbb9ce3c964c350bb9106aef2/packs/snowflake_query_pack.sql#L45) in a Snowflake repo, which suggests all of the following error codes are authorization failures: `(1063, 3001, 3003, 3005, 3007, 3011, 3041)`.

Notice that we already catch and handle 3001 on line 57.  And here you add a catch for 3041.  Could we consolidate those two matches into one matcher?  i.e. amend line 57 to catch 3001 OR 3041 (and possibly the other linked error codes too).

Reply:
I suggest that we have two different checks (two `case`), one on the error codes, and one on the error message only. This way if error code or message changes, we still catch the error